File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        int n = s.length();
        int left = 0;
        int right = n-1;
        long long count = 0;
        while(left<right){
            while(s[left]=='0' && right>left){
                left++;
            }
            while(s[right]=='1' && right>left){
                right--;
            }
            
            if(s[left]=='1' && right>left){
                count += right-left;
                left++;
                right--;
            }
        }  
        return count;      
    }
};

File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.go
func minimumSteps(s string) int64 {
    var count int64 = 0
    left:=0
    n:=len(s) 
    right:=n-1
    for right>left{
        for s[left]=='0' && left<right{
            left++
        }

        for s[right]=='1' && left<right{
            right--
        }

        if s[left]=='1' && right>left{
            count+=int64(right-left)
            right--
            left++
        }

    }
    return count;
}